app/controllers/conferences_controller.rb

```diff
@@ -6,6 +6,9 @@ class ConferencesController < ApplicationController
 
   def index
     @current = Conference.where('end_date >= ?', Date.current).reorder(start_date: :asc)
+    if @current.count == 1
+      redirect_to "/conferences/#{@current.first.id}"
```

Comment:
Yes @shlok007 it does not work, as routing is based on short_title, also when I the URL `/conferences/:id` with the short_title in `params[:id]`, it redirects me back to homepage saying to login and even after login in it only works for the admin, any user or non-user cannot acess the conference#show show authorization needs to be relaxed in this case.